source/ranges.tex

```diff
@@ -3802,1398 +3855,2470 @@
-\tcode{pattern_} with \tcode{std::move(pattern)}.
+\effects Equivalent to: \tcode{return x.i_.current_ == y.i_.current_;}
 \end{itemdescr}
 
-\indexlibrary{\idxcode{split_view}!\idxcode{split_view}}%
+\indexlibrary{\idxcode{operator"!=}!\idxcode{split_view::inner_iterator}}%
 \begin{itemdecl}
-template<InputRange R, ForwardRange P>
-  requires Constructible<V, all_view<R>> &&
-           Constructible<Pattern, all_view<P>>
-constexpr split_view(R&& r, P&& p);
+friend constexpr bool operator!=(const inner_iterator& x, const inner_iterator& y)
+  requires ForwardRange<Base>;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-Initializes \tcode{base_} with \tcode{view::all(std::forward<R>(r))} and
-\tcode{pattern_} with \tcode{view::all(std\brk{}::forward<P>(p))}.
+\effects Equivalent to: \tcode{return !(x == y);}
 \end{itemdescr}
 
-\indexlibrary{\idxcode{split_view}!\idxcode{split_view}}%
+\indexlibrary{\idxcode{operator==}!\idxcode{split_view::inner_iterator}}%
 \begin{itemdecl}
-template<InputRange R>
-  requires Constructible<V, all_view<R>> &&
-           Constructible<Pattern, single_view<iter_value_t<iterator_t<R>>>>
-constexpr split_view(R&& r, iter_value_t<iterator_t<R>> e);
+friend constexpr bool operator==(const inner_iterator& x, default_sentinel_t);
+friend constexpr bool operator==(default_sentinel_t, const inner_iterator& x);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-Initializes \tcode{base_} with \tcode{view::all(std::forward<R>(r))} and
-\tcode{pattern_} with \tcode{single_view\{\brk{}std::move(e)\}}.
+\effects Equivalent to:
+\begin{codeblock}
+auto cur = x.i_.@\placeholder{current}@;
+auto end = ranges::end(x.i_.parent_->base_);
+if (cur == end) return true;
+auto [pcur, pend] = subrange{x.i_.parent_->pattern_};
+if (pcur == pend) return x.incremented_;
+do {
+  if (*cur != *pcur) return false;
+  if (++pcur == pend) return true;
+} while (++cur != end);
+return false;
+\end{codeblock}
 \end{itemdescr}
 
-\rSec3[range.split.outer]{Class template \tcode{split_view::outer_iterator}}
+\indexlibrary{\idxcode{operator"!=}!\idxcode{split_view::inner_iterator}}%
+\begin{itemdecl}
+friend constexpr bool operator!=(const inner_iterator& x, default_sentinel_t y);
+friend constexpr bool operator!=(default_sentinel_t y, const inner_iterator& x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to: \tcode{return !(x == y);}
+\end{itemdescr}
+
+\indexlibrary{\idxcode{iter_swap}!\idxcode{split_view::inner_iterator}}%
+\begin{itemdecl}
+friend constexpr void iter_swap(const inner_iterator& x, const inner_iterator& y)
+  noexcept(noexcept(ranges::iter_swap(x.i_.@\placeholdernc{current}, y.i_.\placeholdernc{current}@)))
+  requires IndirectlySwappable<iterator_t<Base>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to
+\tcode{ranges::iter_swap(x.i_.\placeholdernc{current}, y.i_.\placeholdernc{current})}.
+\end{itemdescr}
+
+\rSec3[range.split.adaptor]{\tcode{view::split}}
+
+\pnum
+The name \tc
```

Comment:
This is good, but other views already use sibling specialization's private members without doing this. I may have mentioned this is some issue.

Reply:
Sorry, I don't know where this comment attaches to.

Reply:
Ok, found that place. What do you want me to do here? Add the friend everywhere, or remove the friend here? Do you have an issue number?

Reply:
It seems I didn't raise anything since it passed LWG review. I don't remember my results from looking into what the WP did before the merging of the One Ranges Proposal, but while I may have found some friend declarations of sibling specializations for the sake of using their private members, finding those that didn't do that would be hard so I didn't go that far. My suggestion would be to remove this friend here for the sake of local consistency, and maybe to file an issue for the general case.

Reply:
This is touching normative content, it seems, so I'll not do this. Please file an LWG issue for guidance / cleanup.